Add `Yours` issue/PR search tab

busches opened the pull request:
Add a 'Yours' menu tab to the top navigation. This does a Github search
for all open items created by yourself, which includes PRs and Issues.
Fixes #183 

Shortcut is 'g y' and appears after Issues in the menu.

jgierer12 commented:
You can also take a look at #228 where @jacobbearden already worked on it.

hkdobrev commented:
👍 On what @jgierer12 said. IMO it should be as discussed in #183 - link on issues page and link on PRs page for a `user:` filter on these pages. The global search interface you are using at the moment is not the same thing and is not as useful. Also, the original issue #183 was not about issues created by you, but issues created on your own repos.

hkdobrev commented:
@busches What do you think about the comments? Could you move it as filters on the issues and PRs pages instead of being a global search?

busches commented:
Yup, just need to get a moment to do it. 😀

On May 12, 2017 4:27 PM, "Haralan Dobrev" <notifications@github.com> wrote:

> @busches <https://github.com/busches> What do you think about the
> comments? Could you move it as filters on the issues and PRs pages instead
> of being a global search?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/sindresorhus/refined-github/pull/406#issuecomment-301190121>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AA0WZHSlFkBcUUBOBBFHGxT5x8BZXKrzks5r5M61gaJpZM4NS_nx>
> .
>

busches commented:
I used what @jacobbearden did in #228 to start with and rebased against the current master. Then I fixed the new/outstanding issues from the PR. Please let me know if you find anything else.

jgierer12 commented on extension/content.js:
```diff
@@ -113,6 +113,24 @@ function addTrendingMenuItem() {
+		return;
+	}
+
+	const yoursMenuItem = $(`<a href="/${page}?q=is%3Aopen+is%3Aissue+user%3A${username}" class="subnav-item">Yours</a>`);
```
Not sure if it makes sense to keep the same page number

busches replied:
It's not the page number, but instead either `pulls` or `issues`. I'll rename it to make it clearer.

jgierer12 replied:
Oh, that makes sense. Then I suggest you make this a parameter for the function and pass it depending on which of the `pageDetect`s is active

busches replied:
Passing down a string in two different spots seems convoluted compared to checking it inside the function.

This is what I see it changing to:
```
if (pageDetect.isIssueSearch() || pageDetect.isPRSearch()) {
	addYoursMenuItem(pageDetect.isIssueSearch() ? 'issues' : 'pulls');
}
```
Can you think of a cleaner solution than that? As that seems messier than what is existing.

jgierer12 replied:
```js
if (pageDetect.isIssueSearch()) {
    addYoursMenuItem('issues');
}
if (pageDetect.isPRSearch()) {
   addYoursMenuItem('pulls');
}
```
or at least use `pageDetect` to set the `pageName` variable to keep DRY

busches replied:
Updated to use the `pageDetect` functions to set it. Passing a string in four places didn't feel DRY.

jgierer12 commented on extension/content.js:
```diff
@@ -113,6 +113,24 @@ function addTrendingMenuItem() {
+	const username = getUsername();
+	const $menu = $('.subnav-links');
+
+	if ($(`.subnav-item[href="/${page}?q=is%3Aopen+is%3Aissue+user%3A${username}"]`).length > 0) {
```
I'd rather use a dedicated class for this (by adding a `refined-github-yours-tab` class to the button and checking for that)

sindresorhus commented:
> Note: the 'Yours' on the PR page returns me no results, not sure if the search is wrong or I don't have anything applicable, I would expect this PR to be in that list though, which it is under created.

The search is incorrect. `is:open is:pr user:sindresorhus ` => `is:open is:pr author:sindresorhus`

jgierer12 commented:
> The 'Yours' on the PR page returns me no results.

Works fine for me

> I would expect this PR to be in that list though, which it is under created

No, "Created" lists everything that you posted. "Yours" lists everything that was posted on one of your own repos

sindresorhus commented:
It doesn't fit for me, since I have "Review Requests":

<img width="788" alt="screen shot 2017-05-22 at 23 59 15" src="https://cloud.githubusercontent.com/assets/170270/26319825/bc876c46-3f4a-11e7-92b3-27eaad214dbd.png">

sindresorhus commented on extension/page-detect.js:
```diff
@@ -11,10 +11,14 @@ window.pageDetect = (() => {
 
 	const isRepoTree = () => isRepo() && /\/tree\//.test(getRepoPath());
 
+	const isIssueSearch = () => location.pathname.indexOf('/issues') === 0;
```
`indexOf` => `startsWith`

busches replied:
This would make for a good rule in `xo`.

sindresorhus replied:
@busches Indeed, but ESLint doesn't know that this is a string, it could be an Array too... JS is cool, but a good type system would have been wonderful.

https://github.com/sindresorhus/eslint-plugin-unicorn/pull/64#issuecomment-269987212

sindresorhus commented on readme.md:
```diff
@@ -34,6 +34,7 @@ Our hope is that GitHub will notice and implement some of these much needed impr
 - [Adds navigation to milestone pages](https://cloud.githubusercontent.com/assets/170270/25217211/37b67aea-25d0-11e7-8482-bead2b04ee74.png)
 - [Adds search filter for 'Everything commented by you'](https://cloud.githubusercontent.com/assets/940070/25518367/cb917d3e-2c36-11e7-8475-c4e6dbe0ed6c.png)
 - [Moves destructive buttons ("Close issue", "Cancel") in commenting forms away from primary button](#comment-box)
+- [Adds `Yours` button to Issues/Pull Reqeusts page](https://cloud.githubusercontent.com/assets/1282980/14636384/0d8770e4-0623-11e6-8520-2054bece2771.png)
```
typo

sindresorhus commented on extension/content.js:
```diff
@@ -113,6 +113,24 @@ function addTrendingMenuItem() {
+
+	const yoursMenuItem = $(`<a href="/${page}?q=is%3Aopen+is%3Aissue+user%3A${username}" class="subnav-item">Yours</a>`);
+
+	if ($('.subnav-links .selected').length === 0 && new RegExp(`user%3A${username}`).test(location.search)) {
```
> new RegExp(`user%3A${username}`).test(location.search)

I don't see why regex is needed here. Just use `String#includes`.

busches commented:
@jgierer12 @sindresorhus comments addressed, thanks for the feedback.

@sindresorhus I also have `Review Requests` and it fits for me on Windows. Only idea would be to shrink the search input box, would that work? If yes, can you tell me what size works for you, so I don't have to guess? :)

jgierer12 commented:
> I also have Review Requests and it fits for me on Windows

It fits for me too, but it depends on screen size of course. You could just remove the `subnav-search-input-wide` class on the search box

busches commented:
@jgierer12 Passing a string down seems convoluted when we can get it out of the URL and only modify one spot, instead of having to check if we should call the function, then pass the string down in multiple spots. If you can think of a cleaner way to implement it then that, I'm open to it.

busches commented:
Removed the `subnav-search-input-wide` class, the box does look quite small now.

sindresorhus commented:
@busches You could add `.subnav-search-input[aria-label="Search all issues"] { width: 420px; }` to make it wider.

busches commented:
@sindresorhus Making it `420` looks a lot better, done.

sindresorhus commented:
Awesome. Looks great now! :)
